pkg/tenant/members.go

```diff
@@ -261,6 +262,105 @@ func (s *Server) MemberUpdate(c *gin.Context) {
 	c.JSON(http.StatusOK, m.ToAPI())
 }
 
+func (s *Server) MemberRoleUpdate(c *gin.Context) {
+	var err error
+
+	// Members exist on organizations
+	// This method handles the logging and error responses
+	var orgID ulid.ULID
+	if orgID = orgIDFromContext(c); ulids.IsZero(orgID) {
+		return
+	}
+
+	// Get the member ID from the URL and return a 400 if the member ID is not a ULID.
+	var memberID ulid.ULID
+	if memberID, err = ulid.Parse(c.Param("memberID")); err != nil {
+		sentry.Warn(c).Err(err).Str("id", c.Param("memberID")).Msg("could not parse member id")
+		c.JSON(http.StatusNotFound, api.ErrorResponse("member not found"))
+		return
+	}
+
+	// TODO: Add org verification
+
+	// Bind the user request with JSON.
+	params := &api.UpdateMemberParams{}
+	if err = c.BindJSON(&params); err != nil {
+		sentry.Warn(c).Err(err).Msg("could not parse member update request")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(api.ErrUnparsable))
+		return
+	}
+
+	// Verify member role exists.
+	if params.Role == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("member role is required"))
+		return
+	}
+
+	// Verify the role provided is valid.
+	perms.IsRole(params.Role)
```

Comment:
Sorry, should have mentioned that this method just returns a bool so we do need to check the result and return a `StatusBadRequest` if necessary.